src/Microsoft.OpenApi.Readers/OpenApiDocumentExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Readers;
+
+namespace Microsoft.OpenApi
+{
+    /// <summary>
+    /// Loads an OpenApiDocument instance through Load/LoadAsync/Parse pattern
+    /// </summary>
+    public static class OpenApiDocumentExtensions
```

Comment:
Maybe rename this class to `OpenApiDocumentReaderExtensions`

Reply:
The current class implementation extends the OpenApiDocument object, not the document readers hence the name. The only issue is that we can't add it to the models lib due to the circular dependency issue.